transformers-compat.cabal

```diff
@@ -96,31 +101,45 @@ library
   other-modules:
     Paths_transformers_compat
 
+  -- automatic flags
+  if flag(five-three)
+    hs-source-dirs: 0.5
+    build-depends: transformers >= 0.5.3 && < 0.6
+  else
+    build-depends: transformers < 0.5.3
+
   if flag(five)
     hs-source-dirs: 0.5
     build-depends: transformers >= 0.5 && < 0.5.3
   else
-    if flag(four)
```

Comment:
What is the rationale for chopping up this part of the `.cabal` file like you've done here?

Reply:
if there is a cascade of flags

    if flag(foo)
      somethingX
    else
      if flag(bar)
        somethingY
      else
        somethingZ

then `+foo +bar` and `+foo -bar` configurations are the same: flag selection is ambigious.

If however we have

    if flag(foo)
      somethingX
    else
      somethingNotX

    if flag(bar)
      somethingY
    else
      somethingNotY

then all four combinations `+foo +bar`, `+foo -bar`, `-foo +bar` and `-foo -bar` are different.

---

I set flags in a way we should have single flag on (actually have to check that all-off case is invalid).
There's non-converged discussion how to make that more sane: https://github.com/haskell/cabal/issues/3526